crates/rustc_codegen_spirv/src/codegen_cx/constant.rs

```diff
@@ -374,23 +386,88 @@ impl<'tcx> CodegenCx<'tcx> {
+                    ),
+                    SpirvType::Integer(int_size, int_signedness) => {
+                        let integer = match int_size {
+                            8 => Integer::I8,
+                            16 => Integer::I16,
+                            32 => Integer::I32,
+                            64 => Integer::I64,
+                            128 => Integer::I128,
+                            other => {
+                                self.tcx
+                                    .sess
+                                    .fatal(&format!("invalid size for integer: {}", other));
+                            }
+                        };
+                        (
+                            Primitive::Int(integer, int_signedness),
+                            SpirvType::Integer(int_size, int_signedness)
+                                .sizeof(self)
+                                .unwrap(),
+                        )
+                    }
+                    SpirvType::Float(float_size) => {
+                        let float_size_primitive = match float_size {
+                            32 => Primitive::F32,
+                            64 => Primitive::F64,
+                            other => {
+                                self.tcx
+                                    .sess
+                                    .fatal(&format!("invalid size for float: {}", other));
+                            }
+                        };
+                        (
+                            float_size_primitive,
+                            SpirvType::Float(float_size).sizeof(self).unwrap(),
+                        )
+                    }
+                    SpirvType::Pointer {
+                        storage_class,
+                        pointee,
+                    } => (
+                        Primitive::Pointer,
+                        SpirvType::Pointer {
+                            storage_class,
+                            pointee,
+                        }
+                        .sizeof(self)
+                        .unwrap(),
+                    ),
+                    unsupported_spirv_type => bug!(
+                        "invalid spirv type internal to create_alloc_const2: {:?}",
+                        unsupported_spirv_type
+                    ),
+                };
+                let scalar = match alloc
+                    .read_scalar(self, Pointer::new(AllocId(0), *offset), size)
+                    .unwrap()
+                {
+                    ScalarMaybeUninit::Scalar(scalar) => scalar,
+                    ScalarMaybeUninit::Uninit => self
                         .tcx
                         .sess
-                        .fatal(&format!("invalid float width {}", other)),
-                }
+                        .fatal("allocation pointed at uninitialized data"),
+                };
+                let scalar_unit = |value: Primitive| {
```

Comment:
why is this a lambda? why not straight code?

Reply:
Good call, the source was a lambda and I never cleaned it up. Will do